src/udp/rendezvous_server.rs

```diff
@@ -33,7 +33,12 @@ impl UdpRendezvousServer {
             .unwrap_or(UDP_RENDEZVOUS_PORT);
         let addr = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), port));
         let sock = UdpSock::bind(&addr)?;
+        Self::start_with_sock(sock, ifc, poll)
+    }
 
+    /// Boot the UDP rendezvous server and use the given UDP socket to listen for incoming
+    /// requests.
+    pub fn start_with_sock(sock: UdpSock, ifc: &mut Interface, poll: &Poll) -> ::Res<Token> {
```

Comment:
Similar to another comment in this review. I'm not comfortable exposing stuff only from a crate only for the purposes of testing. The crate should not become a "debug-library" where all the internals are exposed to the user and not for any non-trivial reason. This function conflicts with the config file usage and the design does not make much sense when things can be tweaked in several different ways without each ignoring the other specifications. In this case for e.g., i would rather use an in-memory config file which can easily be created per test and different to other tests, to bind to some specific port etc.